core/libdeno/api.cc

```diff
@@ -233,4 +233,11 @@ void deno_terminate_execution(Deno* d_) {
   deno::DenoIsolate* d = reinterpret_cast<deno::DenoIsolate*>(d_);
   d->isolate_->TerminateExecution();
 }
+
+void deno_run_microtasks(Deno* d_) {
```

Comment:
I suspect you'll have to update the tests in libdeno_test.cc

Reply:
We may need a user_data here and a UserDataScope within this function.

Reply:
i agree

Reply:
Just tested, and this does fix most of the tests.

```
diff --git a/core/isolate.rs b/core/isolate.rs
index 99c3ee06..d555ce7b 100644
--- a/core/isolate.rs
+++ b/core/isolate.rs
@@ -390,7 +390,7 @@ impl Isolate {
   }
 
   fn run_microtasks(&self) -> Result<(), ErrBox> {
-    unsafe { libdeno::deno_run_microtasks(self.libdeno_isolate) };
+    unsafe { libdeno::deno_run_microtasks(self.libdeno_isolate, self.as_raw_ptr()) };
     self.check_last_exception()
   }
 
diff --git a/core/libdeno.rs b/core/libdeno.rs
index d8a6d771..65d4cc28 100644
--- a/core/libdeno.rs
+++ b/core/libdeno.rs
@@ -280,7 +280,7 @@ extern "C" {
     js_source: *const c_char,
   );
   pub fn deno_terminate_execution(i: *const isolate);
-  pub fn deno_run_microtasks(i: *const isolate);
+  pub fn deno_run_microtasks(i: *const isolate, user_data: *const c_void);
 
   // Modules
 
diff --git a/core/libdeno/api.cc b/core/libdeno/api.cc
index 2df40b21..2e926f79 100644
--- a/core/libdeno/api.cc
+++ b/core/libdeno/api.cc
@@ -234,8 +234,10 @@ void deno_terminate_execution(Deno* d_) {
   d->isolate_->TerminateExecution();
 }
 
-void deno_run_microtasks(Deno* d_) {
+void deno_run_microtasks(Deno* d_, void* user_data) {
   deno::DenoIsolate* d = reinterpret_cast<deno::DenoIsolate*>(d_);
+
+  deno::UserDataScope user_data_scope(d, user_data);
   v8::Locker locker(d->isolate_);
   v8::Isolate::Scope isolate_scope(d->isolate_);
   d->isolate_->RunMicrotasks();
diff --git a/core/libdeno/deno.h b/core/libdeno/deno.h
index bccb4b57..0bdd31f5 100644
--- a/core/libdeno/deno.h
+++ b/core/libdeno/deno.h
@@ -119,7 +119,7 @@ const char* deno_last_exception(Deno* d);
 
 void deno_terminate_execution(Deno* d);
 
-void deno_run_microtasks(Deno* d);
+void deno_run_microtasks(Deno* d, void* user_data);
 // Module API
 
 typedef int deno_mod;
```

Still a few others to figure out though mainly in `libdeno_test`. I ran a quick benchmark, and if the intent was to improve performance you might be a little disappointed with this one.